src/picongpu/include/fields/background/templates/TWTS/RotateField.tpp

```diff
@@ -0,0 +1,127 @@
+                fieldPosVector.x(),
+               -pmMath::sin(AngleType(phi))*fieldPosVector.y()
+                    -pmMath::cos(AngleType(phi))*fieldPosVector.z() ,
+               +pmMath::cos(AngleType(phi))*fieldPosVector.y()
+                    -pmMath::sin(AngleType(phi))*fieldPosVector.z() );
+        }
+                        
+    };
+    
+    template <typename T_Type, typename T_AngleType>
+    struct RotateField<PMacc::math::Vector<T_Type,2>, T_AngleType >
+    {
+        typedef PMacc::math::Vector<T_Type,2> result;
+        typedef T_AngleType AngleType;
+        HDINLINE result
+        operator()( const result& fieldPosVector,
+                    const AngleType phi ) const
+        {
+            /*  Since, the laser propagation direction encloses an angle of phi with the
+             *  simulation y-axis (i.e. direction of sliding window), the positions vectors are
+             *  rotated around the simulation x-axis before calling the TWTS field functions.
+             *  Note: The TWTS field functions are in non-rotated frame and only use the angle
+             *  phi to determine the required amount of pulse front tilt.
+             *  RotationMatrix[PI/2+phi].(y,z) (180Deg-flip at phi=90Deg since coordinate
+             *  system in paper is oriented the other way round.) */
+            
+            /*  Rotate 90 degree around y-axis, so that TWTS laser propagates within
+             *  the 2D (x,y)-plane. Corresponding position vector for the Ez-components
+             *  in 2D simulations.
+             *  3D     3D vectors in 2D space (x,y)
+             *  x -->  z
+             *  y -->  y
+             *  z --> -x (Since z=0 for 2D, we use the existing
+             *            TWTS-field-function and set -x=0)
+             *  Ex --> Ez (--> Same function values can be used in 2D,
+             *             but with Yee-Cell-Positions for Ez.)
+             *  By --> By
+             *  Bz --> -Bx
+             *
+             * Explicit implementation in 3D coordinates:
+             * fieldPosVector = float3_64( -fieldPosVector.z(),       //(Here: ==0)
+             *                              fieldPosVector.y(),
+             *                              fieldPosVector.x() );
+             * fieldPosVector = float3_64( fieldPosVector.x(),
+             *       -sin(phi)*fieldPosVector.y()-cos(phi)*fieldPosVector.z(),
+             *       +cos(phi)*fieldPosVector.y()-sin(phi)*fieldPosVector.z()  );
+             * The 2D implementation here only calculates the last two components.
+             * Note: The x-axis of rotation is fine in 2D, because that component now contains
+             *       the (non-existing) simulation z-coordinate. */
+             return result(
+                -pmMath::sin(AngleType(phi))*fieldPosVector.y()
+                    -pmMath::cos(AngleType(phi))*fieldPosVector.x() ,
+                +pmMath::cos(AngleType(phi))*fieldPosVector.y()
```

Comment:
Readability. Also it shows that among the 3 `-` signs it is not a typo, but intended.